src/Aspire.Hosting.MongoDB/MongoDBBuilderExtensions.cs

```diff
@@ -15,22 +17,57 @@ public static class MongoDBBuilderExtensions
     // Internal port is always 27017.
     private const int DefaultContainerPort = 27017;
 
+    private const string UserEnvVarName = "MONGO_INITDB_ROOT_USERNAME";
+    private const string PasswordEnvVarName = "MONGO_INITDB_ROOT_PASSWORD";
+
     /// <summary>
     /// Adds a MongoDB resource to the application model. A container is used for local development. This version the package defaults to the 7.0.8 tag of the mongo container image.
     /// </summary>
     /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
     /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
     /// <param name="port">The host port for MongoDB.</param>
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
-    public static IResourceBuilder<MongoDBServerResource> AddMongoDB(this IDistributedApplicationBuilder builder, string name, int? port = null)
+    public static IResourceBuilder<MongoDBServerResource> AddMongoDB(this IDistributedApplicationBuilder builder, string name, int? port)
     {
-        var mongoDBContainer = new MongoDBServerResource(name);
+        return AddMongoDB(builder, name, port, null, null);
+    }
 
-        return builder
+    /// <summary>
+    /// Adds a MongoDB resource to the application model. A container is used for local development. This version the package defaults to the 7.0.8 tag of the mongo container image.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="port">The host port for MongoDB.</param>
+    /// <param name="userName">A parameter that contains the MongoDb server user name, or <see langword="null"/> to use a default value.</param>
+    /// <param name="password">A parameter that contains the MongoDb server password, or <see langword="null"/> to use a generated password.</param>
+    /// <param name="authenticationMechanism">A parameter that contains the MongoDb server authentication mechanism, or <see langword="null"/> to use a default value.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<MongoDBServerResource> AddMongoDB(this IDistributedApplicationBuilder builder,
+        string name,
+        int? port = null,
+        IResourceBuilder<ParameterResource>? userName = null,
+        IResourceBuilder<ParameterResource>? password = null,
+        IResourceBuilder<ParameterResource>? authenticationMechanism = null)
```

Comment:
Do we need this authentication mechanism as a public option? I don't see it in other resources - ex. PostgreSQL.

Reply:
We don't